woocommerce/woocommerce
# Fix payment gateway suggestions display logic

### Changes proposed in this Pull Request:

This PR
- Fixes payment gateway display logic. Previously, it was displaying "other" gateways for WCPay-ineligible countries even when it's not defined in the "other" category.
- Update tests to reflect the correct logic.
- Added rule to support offline selling venues to `woocommerce_onboarding_profile`
- Added `IN` to PayPal's visibility rule.
- Updates variable names and comments to reduce confusion (hopefully)



### How to test the changes in this Pull Request:

**Enable core profiler feature**
1. Check out this branch and [build the plugin](https://github.com/woocommerce/woocommerce/blob/trunk/DEVELOPMENT.md).
2. Download and install the [WooCommerce Beta Tester](https://wordpress.org/plugins/woocommerce-beta-tester/) plugin.
3. In `wp-admin`, under **Tools > WCA Test Helper**, click the Features link in the top menu.
4. Find the `core-profiler` feature flag and enable it via the toggle.

**Testing new onboarding option rule**
1. Go to OBW
3. In "Which one of these best describes you?",  Select `I'm already selling` and `No, I'm selling offline`
1. Choose Japan as the store country
1. Complete OBW without installing any plugins in the business step.
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` > Untick "Show Suggestions" checkbox
1. Go to `WooCommerce -> Home` and choose Set up payments task.
1. Observe that `Stripe`, `PayPal Payments` and `Square` are displayed in order.

**Testing Paypal in India**
1. Change store country to India
1. Observe that Paypal doesn't show up
